protecode/util.py

```diff
@@ -106,23 +154,69 @@ def iter_artefact_metadata(
                 data=license_finding,
                 discovery_date=discovery_date,
             )
-            has_license_findings = True
 
-        meta = dso.model.Metadata(
-            datasource=dso.model.Datasource.BDBA,
-            type=dso.model.Datatype.VULNERABILITY,
-            creation_date=now,
-            last_update=now,
-        )
+            # remove license finding because it is still relevant and must not be deleted afterwards
+            existing_findings = [
+                existing_finding for existing_finding in existing_findings
+                if not (
+                    existing_finding.meta.type == dso.model.Datatype.LICENSE
+                    and existing_finding.data.license.name == license.name
+                    and existing_finding.data.id.package_name == package_id.package_name
+                    and existing_finding.data.id.package_version == package_id.package_version
+                )
+            ]
 
         for vulnerability in package.vulnerabilities():
-            if (
-                vulnerability.historical() or
-                vulnerability.has_triage() or
-                not vulnerability.cvss
-            ):
+            if not vulnerability.cvss:
+                # we only support vulnerabilities with a valid cvss v3 vector
+                continue
+
+            if vulnerability.historical():
                 continue
 
+            for triage in vulnerability.triages():
+                meta = dso.model.Metadata(
+                    datasource=dso.model.Datasource.BDBA,
+                    type=dso.model.Datatype.RESCORING,
+                    relation=dso.model.Relation(
+                        refers_to=dso.model.Datatype.VULNERABILITY,
+                        relation_kind=dso.model.RelationKind.RESCORE,
+                    ),
+                    creation_date=triage.modified.astimezone(datetime.UTC),
+                )
+
+                vulnerability_rescoring = dso.model.CustomRescoring(
+                    finding=dso.model.RescoringVulnerabilityFinding(
+                        id=dso.model.BDBAPackageId(
+                            package_name=package_id.package_name,
+                            # don't specify version to store only one rescoring per package name
```

Comment:
should we not rather honour triages' scopes (app vs group, ..) and translate those into our scoping?

Reply:
I thought about this too. However, I came to the conclusion _not_ to do it because
- we can't actually translate them 100% correctly since the BDBA scopes are based on a different model (groups, apps, ...). We _could_ of course translate them in the way it fits Gardener, however, if someone utilises the BDBA groups and so on differently, this mapping wouldn't fit anymore.
- BDBA includes the triages in every scan result which matches its scope. So, if we just convert it to "our" rescoring for every scan and set it to our `single` scope (which means it just applies to the exact tuple of component-id and artefact-id), we basically copy the BDBA scope because we mirror where the rescoring is applied to. The only downside I see to this is that we store one rescoring artefact per version this way, but I think that is okay-ish.
---
So, I would prefer to always use the `single` scope for BDBA triages and import them for every scan result separately.

Reply:
good point - this way, we need less logic on our end and the acceptable price of redundant data.

maybe you could add a comment explaining why we decided against interpreting the scope from bdba